File: mailblastr-python/mailblastr/polls.py
"""Polls resource — read-only results of the in-email poll widget."""

from . import http_client
from .http_client import paginate, path_escape as _e


class Polls:
    """``mailblastr.Polls`` — the /polls endpoints (read-only)."""

    @classmethod
    def list(cls, params=None):
        """One summary row per email that has poll responses. GET /polls"""
        return http_client.request("GET", f"/polls{paginate(params)}")

    @classmethod
    def get(cls, email_id):
        """The aggregated answer breakdown for one email. GET /polls/:email_id"""
        return http_client.request("GET", f"/polls/{_e(email_id)}")

File: mailblastr-java/src/main/java/com/mailblastr/resources/Polls.java
package com.mailblastr.resources;

import com.mailblastr.ListParams;
import com.mailblastr.MailblastrResponse;
import com.mailblastr.http.ApiClient;

/** Read-only results of the in-email poll widget — {@code mailblastr.polls()}. */
public final class Polls extends Resource {
    public Polls(ApiClient api) { super(api); }

    /** One summary row per email that has poll responses. {@code GET /polls} */
    public MailblastrResponse list() { return list(null); }

    public MailblastrResponse list(ListParams params) {
        return api.request("GET", "/polls" + paginate(params));
    }

    /** The aggregated answer breakdown for one email. {@code GET /polls/:emailId} */
    public MailblastrResponse get(String emailId) {
        return api.request("GET", "/polls/" + enc(emailId));
    }
}
